### PR TITLE
docs: API documentation

### PR DESCRIPTION
**Pull request by dietersommer**
### Guides

* Ensure that you have read the [Contributor Guide](blob/main/CONTRIBUTING.md)
* Ensure that you have read the and agreed to the [Code of Conduct](blob/main/CODE_OF_CONDUCT.md)
* Ensure that the title complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: iceth-API.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# IC Eth API
+
+## Overview
+
+*Service, JSON API service:* A Web2 service such as [Infura](https://www.infura.io/), [Gateway.fm](https://gateway.fm/), or [CloudFlare](https://www.cloudflare.com/en-gb/web3/) that offers access to the Ethereum JSON RPC API through HTTP. Note that also other EVM-compatible chains may be covered by such JSON RPC API.
```
Comment (jplevyak) [inline iceth-API.md:5]:
```suggestion
*Service, JSON API service:* A Web2 service such as [Infura](https://www.infura.io/), [Gateway.fm](https://gateway.fm/), or [CloudFlare](https://www.cloudflare.com/en-gb/web3/) that offers access to the Ethereum JSON RPC API through HTTP. Note that also other EVM-compatible chains may be covered by such a JSON RPC API.
```

File: iceth-API.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        cycles_per_call: nat64;
+        cycles_per_message_byte: nat64;
+    };
+
+    register_provider: (RegisterProvider) -> ();
+
+The `RegisterProvider` record defines the details about the service to register, including the API key for the service.
+* `chain_id`: The id of the Ethereum chain this provider allows to connect to. The ids refer to the chain ids as defined for EVM-compatible blockchains, see, e.g., [ChainList](https://chainlist.org/?testnets=true).
+* `service_url`: The URLs of the Web2 service provider that is used by the canister when using this provider.
+* `api_key`: The API key for authorizing requests to this service provider.
```
Comment (jplevyak) [inline iceth-API.md:32]: We should add that the API is private to the registerer and the canister and is not return in by the get_providers() call.  The complete URL used to access the service is constructed by concatenating the service_url and the api_key (without a seperator) e.g. "https://cloudflare-eth.com" and "/my-api-key").

File: iceth-API.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+Clients of this canister need to select a provider that matches w.r.t. the `chain_id` the network they intend to connect to. If multiple providers are available for a given `chain_id`, the per-message or per-byte price or the entity behind the provider (this can be inferred from the `service_url`) may be factors to choose a suitable provider.
+
+### json_rpc_request
+
+Make a request to a Web2 Ethereum node using the caller's URL to an openly available JSON RPC API service, or the caller's URL including an API key for an access-protected API provider. No registered API key of the canister is used in this scenario.
+
+    json_rpc_request: (json_rpc_payload: text, service_url: text, max_response_bytes: nat64) -> (EthRpcResult);
+
+* `json_rpc_payload`: The payload for the JSON RPC request, in compliance with the [JSON RPC specification](https://www.jsonrpc.org/specification).
+* `service_url`: The URL of the service, comprising an API key for access-protected services.
```
Comment (jplevyak) [inline iceth-API.md:71]: Perhaps "including any API key if required" or somesuch.

**Review by jplevyak — APPROVED** [review]

File: iceth-API.md (hunk 2 above)
Comment (dietersommer) [reply iceth-API.md:32]: Good points!

File: iceth-API.md (hunk 3 above)
Comment (dietersommer) [reply iceth-API.md:71]: Done.

**Pull request merged**